network/p2p/impl.go

```diff
@@ -300,18 +282,10 @@ func (n adapter) ConnectToPeer(address string) bool {
 	return false
 }
 
-func (n *adapter) startSendingAndReceiving(conn *connection) {
-	go conn.sendMessages()
-	n.registerConnection(conn)
-	conn.receiveMessages(n.receivedMessages)
-	// When we reach this line, receiveMessages has exited which means the connection has been closed.
-	n.unregisterConnection(conn)
-}
-
 // connectToNewPeers reads from connectorAddChannel to start connecting to new peers
 func (n *adapter) connectToNewPeers() {
```

Comment:
You probably want to migrate this to the connectionManager in the future.

Reply:
It might. I'm not entirely sure how the design will end up at this point, because it risks connectionManager becomes the God object that `adapter` was. I think OK-ish for now, will have to draw diagram to work out the exact components and reponsibilities. (adapter, connector, connection, connectionManger)